Step4

LarryJung opened the pull request:
step4구현하였습니다. 
질문 :
1. 패키지를 domain / view 가 아니라, controller, model, view로 나누어봤는데 domain / view로 다시 나누는 게 좋을까요? 
2. 재귀함수가 들어가있는 부분은 exception테스트를 할 때 재귀문때문에 테스트가 안끝나고 계속 반복되는 나왔습니다.(LadderController의 initPlayers(), initSizeLadder()) 
하지만 이 부분에 들어가 있는 하위 로직들에 대해서는 테스트를 했기 때문에 위 메소드까지 테스트를 해야 하는지는 잘 모르겠습니다.
3. 사용자 예외를 생성하고 로직을 구현하면서 throws를 안해도 상위 메소드에서 에러를 처리할 수 있었는데, 이런 경우에 throws를 명시하지 않고 넘어가는게 맞는건지 궁금합니다.

LarryJung committed:
function add(max name length 5)

LarryJung committed:
function fix(ResultView methods's return type changed, for unit test)

LarryJung committed:
junit test code add

javajigi commented on src/main/java/codesquad/ladder/model/Player.java:
```diff
@@ -1,21 +1,17 @@
     private String name;
 
-    private int length;
+    public Player(){ };
```
빈 생성자가 필요한가?

javajigi commented on src/main/java/codesquad/ladder/model/exceptions/InvalidNameSizeException.java:
```diff
@@ -0,0 +1,4 @@
+package codesquad.ladder.model.exceptions;
+
+public class InvalidNameSizeException extends InvalidPlayerNameException {
```
추후 Exception을 연습하는 곳이 있는데 모두 공부했네요 👍

javajigi commented on src/test/java/codesquad/ladder/controller/LadderControllerTest.java:
```diff
@@ -0,0 +1,153 @@
+    @Test(expected = InvalidPlayerNameException.class)
+    public void playerNameCheckReturnTest2() {
+        String[] playerNames = "pobi,cronggg,honux,jk".split(",");
+        ArrayList<String> names = new ArrayList<String>(Arrays.asList(playerNames));
```
`List<String> names = Arrays.asList("pobi", "cronggg", "honux", "jk");`와 구현할 수 있음.
List는 인터페이스라 다음 미션에서 학습해야 하는데 지금 같이 학습해도 좋겠네요.
다른 부분도 같이 바꿔 보세요.

javajigi commented on src/test/java/codesquad/ladder/controller/LadderControllerTest.java:
```diff
@@ -0,0 +1,153 @@
+        assertEquals(sizeLadder, LadderController.sizeLadderValidCheckReturn(sizeLadder));
+        ;
+    }
+
```
테스트 정말 꼼꼼하게 잘 했네요 💯

javajigi commented on src/main/java/codesquad/ladder/controller/LadderController.java:
```diff
@@ -0,0 +1,117 @@
+import java.util.HashSet;
+import java.util.Set;
+
+public class LadderController {
```
controller 패키지를 두지 않았던 이유는 main() 메소드가 Controller 역할을 한다고 생각했기 때문..

javajigi commented on src/main/java/codesquad/ladder/controller/LadderController.java:
```diff
@@ -0,0 +1,117 @@
+        int numPeople = this.players.size();
+        System.out.println("Put size of Ladder\n>>");
+        int sizeLadder = initSizeLadder(InputView.getNumber());
+        this.ladder = new Ladder(numPeople, sizeLadder);
```
이 부분이 main() 메소드로 간다면..
생성자에서 값을 입력받는 로직이 있는 건 뭔가 좀 찜찜함.

javajigi commented on src/main/java/codesquad/ladder/controller/LadderController.java:
```diff
@@ -0,0 +1,117 @@
+
+    // 사용자 이름 중복 최종 체크
+    static boolean duplicatedNameCheck(ArrayList<String> names) {
+        Set<String> nameSet = new HashSet<String>();
```
오호 벌써 Set도.. 👍

javajigi reviewed (changes requested):
구현 깔끔하게 잘 했네요. Controller 구현 부분에 대해서 좀 더 고민해 보면 좋겠네요.

LarryJung committed:
test code fix, use Arrays.asList

LarryJung committed:
main/controller constructure change -
- remove LadderController constructor
init*method return value change -> void
GameStart method add
